app/Components/Profile/StrategyOrder.tsx

```diff
@@ -0,0 +1,230 @@
+  /**
+   * A pressable list component that changes border highlight when selected
+   * @param point The content of the element
+   * @param key A key for the element
+   * @param order Specifies which number element this is
+   * @param selected Determines if this element is selected or not
+   * @param bullet The prefix/bullet to use for the element
+   * @returns A JSX element of a hint strategy
+   */
+  const StrategyHintComponent = (
+    point: string,
+    key: number | string,
+    order: number,
+    selected: boolean,
+    bullet = "•"
+  ) => {
+    let borderColor = "grey";
+    if (selected) {
+      borderColor = "#D9A05B";
+    }
+
+    const updateSelectedElement = (order: number) => {
+      if (selected) {
+        return -1;
+      } else {
+        return order;
+      }
+    };
+
+    return (
+      <TouchableOpacity
+        style={{
+          flexDirection: "row",
+          padding: 5,
+          paddingLeft: 10,
+          marginTop: 5,
+          borderRadius: 10,
+          borderWidth: 3,
+          borderColor: borderColor,
+        }}
+        key={key}
+        onPress={() => setSelectedElement(updateSelectedElement(order))}
+      >
+        <Text style={{ fontSize: 14, color: "#025E73", minWidth: 20 }}>
+          {bullet}
+        </Text>
+        <Text style={{ fontSize: 14, paddingLeft: 5, color: "#025E73" }}>
+          {point}
+        </Text>
+      </TouchableOpacity>
+    );
+  };
+
+  /**
+   * A list of StrategyHintComponents which is generated by looping through all of the strategies
+   * @param strategies An array of all sudoku strategies
+   * @param selected The index of the selected strategy
+   * @returns JSX StrategyHintComponent
+   */
+  const StrategyHintListComponent = (
+    strategies: sudokuStrategyArray,
+    selected: number
+  ) => {
+    const component: React.JSX.Element[] = [];
+
+    for (const strategy of strategies) {
+      const index = strategies.indexOf(strategy) + 1;
+      component.push(
+        StrategyHintComponent(
+          formatOneLessonName(strategy),
+          strategy,
+          index,
+          selected === index,
+          `${index}.`
+        )
+      );
+    }
+    return component;
+  };
+
+  /**
+   * Updates the selected element and strategy order when up or down arrow is pressed.
+   * @param index The index of the currently selected element
+   * @param increment Determines if the selected element should move up or down the list (-1 is up, 1 is down)
+   */
+  const incrementStrategyOrder = (index: number, increment: -1 | 1) => {
+    let incrementString: "up" | "down";
+    if (increment === 1) {
+      incrementString = "down";
+    } else {
+      incrementString = "up";
+    }
+    // exit early to avoid array out-of-bounds errors
+    if (isIncrementButtonDisabled(index, incrementString)) {
+      return;
+    }
+
+    // converting back 0 based index instead of 1 based index
+    index = index - 1;
+
+    [
```

Comment:
nit: this is really clever but I'm not familiar with the syntax so took me a minute to understand, can you add a comment above explaining its swapping values?

Reply:
I left a comment there, good callout!